src/main/java/com/fp3/haras/view/screens/stable/StableCreate.java

```diff
@@ -60,6 +67,145 @@ private String getCocheira() {
         return String.valueOf(boxStable.getSelectedItem()) + spnStable.getValue();
     }
     
+    public void clearFrame() {
+        spnEnter.setModel(new SpinnerDateModel(new Date(), null, null, Calendar.DAY_OF_YEAR));
+        spnLeave.setModel(new SpinnerDateModel(new Date(), null, null, Calendar.DAY_OF_YEAR));
+        DefaultTableModel tm = (DefaultTableModel) tableProds.getModel();
+        novaEstadia = new Estadia();
+        txtFullValue.setText("");
+        txtIndividualValue.setText("");
+        boxStable.setSelectedItem("A");
+        txtTotalValue.setText("");
+        tableProds.setModel(tm);
+        spnStable.setValue(1);
+        novoConsumo.clear();
+        getTipoEstadia = null;
+        tm.setRowCount(0);
+    }
+
+    public final void updateSuggestionBox() {
+        boxAnimal.removeAllItems();
+        String queryAnimal = "SELECT c FROM Animal c WHERE c.isDeleted = false";
+        List<Animal> animal = EntityUtils.select(queryAnimal, Animal.class);
+        for (Animal item : animal) {
+            boxAnimal.addItem(item.getName());
+        }
+        
+        boxConsumo.removeAllItems();
+        boxConsumo.addItem("");
+        consumoGeral.clear();
+        consumoGeral.add("");
+        
+        String queryConsumo = "SELECT c FROM Produto c WHERE c.isDeleted = false";
+        List<Produto> produto = EntityUtils.select(queryConsumo, Produto.class);
+        for (Produto item : produto) {
+            boxConsumo.addItem(item.getName());
+            consumoGeral.add(item);
+        }
+        
+        String queryServicos = "SELECT c FROM Servico c WHERE c.isDeleted = false";
+        List<Servico> servico = EntityUtils.select(queryServicos, Servico.class);
+        for (Servico item : servico) {
+            boxConsumo.addItem(item.getName());
+            consumoGeral.add(item);
+        }
+        
+        boxTipo.removeAllItems();
+        String queryTipo = "SELECT c FROM TipoEstadia c WHERE c.isDeleted = false";
+        List<TipoEstadia> tipo = EntityUtils.select(queryTipo, TipoEstadia.class);
+        for (TipoEstadia item : tipo) {
+            boxTipo.addItem(item.getType());
+            tipoEstadias.add(item);
+        }
+        
+        if (!tipoEstadias.isEmpty() && boxTipo.getSelectedIndex() != -1) {
+            TipoEstadia e = tipoEstadias.get(boxTipo.getSelectedIndex());
+            txtTotalValue.setText(String.valueOf(e.getPrice()));
+        }
+    }
+    
+    private void updateProduto(Produto p) {
+        prodModel = (DefaultTableModel) tableProds.getModel();
+        
+        boolean exists = false;
+        
+        for (int i = 0; i < prodModel.getRowCount(); i++) {
+            if ((int)spnProdUsed.getValue() == 0) {
+                if (prodModel.getValueAt(i, 0).equals(p.getName()) 
+                && !prodModel.getValueAt(i, 1).equals("Tipo Único")) {
```

Comment:
I think it's a good practice to store this text variable in constants. So I would make something in the beginning of the file like this:
```java
private final String type = "Tipo Único";
```
and then utilize that constant where it's needed.